.github/workflows/common-docker.yml

```
@@ -0,0 +1,107 @@
+name: Docker
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: Environment (network) for workflow execution
+        required: false
+        type: string
+      download-artifact-name:
+        required: true
+        type: string
+      download-artifact-path:
+        required: true
+        type: string
+      download-artifact-tar:
+        description: Set to true if provided `download-artifact-path` is compressed tar file
+        required: true
+        default: false
+        type: boolean
+      image-name:
+        description: Docker image name
+        required: true
+        type: string
+      image-push:
+        required: true
+        default: false
+        type: boolean
+      image-context:
+        required: true
+        type: string
+    secrets:
+      gcr-json-key:
+        description: JSON key for Google Container Registry service account (required if image-push is true)
+        required: false
+
+jobs:
+  docker-build-and-publish:
+    name: Build and publish Docker image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/download-artifact@v3
+        if: inputs.download-artifact-tar != true
+        with:
+          name: ${{ inputs.download-artifact-name }}
+          path: ${{ inputs.download-artifact-path }}
+
+      - uses: actions/download-artifact@v3
```

Optional suggestion:
```suggestion
      - name: Download compressed artifacts
      - uses: actions/download-artifact@v3
```